### PR TITLE
feat: Implement phone and email validations

### PR DESCRIPTION
**Pull request by vsgoulart**
Relates camunda/team-hto#52, camunda/team-hto#53

### CODE REVIEW

**Comment by vsgoulart** [comment]
@Skaiir @pinussilvestrus Any idea why the CI build is failing in files I didn't change?

**Comment by vsgoulart** [comment]
I think [this commit](https://github.com/bpmn-io/form-js/commit/c354a614949a599b5b8569f51b12adf755291101) broke `design-sprint` @pinussilvestrus

**Comment by pinussilvestrus** [comment]
We shouldn't aim for the tests to be green this week. (cf. [Logistics](https://docs.google.com/document/d/1WspTuSqqXTltnauSCrLDAZE8S6US1AovS9of7l2bmuo/edit#heading=h.t93goyndpe5u))

**Comment by pinussilvestrus** [comment]
> Closes https://github.com/camunda/team-hto/issues/52, https://github.com/camunda/team-hto/issues/53

Let's not close the tracking issue after this PR. Rationale: we still need to bring these into `develop` + any other task inside the tracking issue.

**Comment by pinussilvestrus** [comment]
> We shouldn't aim for the tests to be green this week. (cf. [Logistics](https://docs.google.com/document/d/1WspTuSqqXTltnauSCrLDAZE8S6US1AovS9of7l2bmuo/edit#heading=h.t93goyndpe5u))

The only thing that should work is `npm run build`, that's needed for the demo.

File: packages/form-js-viewer/src/core/Validator.js [hunk 1]
```diff
@@ -54,6 +58,20 @@ export default class Validator {
       ];
     }
 
+    if ('validationType' in validate && value && validate.validationType === 'phone' && !PHONE_PATTERN.test(value)) {
+      errors = [
+        ...errors,
+        'Field must be a valid phone number.'
```
Comment (pinussilvestrus) [inline packages/form-js-viewer/src/core/Validator.js:64]: Should maybe give a hint in the error message about what a valid phone number looks like? Might be ok for the demo though.

image

**Review by pinussilvestrus — COMMENTED** [review]
Looks good for the demo 👍

**Pull request merged**